Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -1,26 +1,87 @@
+import { useState } from "react";
+
 import BatteryInput from "~/components/batteryInput";
+import CalculateButton from "~/components/calculateButton";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
+import ErrorMessage from "./components/errorMessage";
+import RangeDisplay from "./components/rangeDisplay";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
+  const calculateRange = (s: number, b: number, w:number) => {
+    return -(s * s * b / 2500) + (4 * b) + w;
+  };
+
+  // Set up hooks for form field states
+  const [speed, setSpeed] = useState("");
+  const [battery, setBattery] = useState("");
+  const [weather, setWeather] = useState(50);
+  const [display, setDisplay] = useState(false);
+
+  // Set up hooks for error states
+  // Going to set error flags as false so errors don't display at first render when fields are empty
+  const [isSpeedEmpty, setSpeedEmpty] = useState(false);
+  const [isBatteryEmpty, setBatteryEmpty] = useState(false);
+  const [isSpeedError, setSpeedError] = useState(false);
+  const [isBatteryError, setBatteryError ] = useState(false);
```

Comment:
Why do we need to use state to represent the errors here? Think about it this way, if this teaser had 50 input fields, would you want to create 50 state variables to represent the error for each one?